### DISCUSSION TITLE
Micronaut 4: Expression in @Scheduled is not resolved at runtime

### DISCUSSION THREAD

**Discussion by saw303** [micronaut-projects/micronaut-core#9594] [Q&A]
I'm migrating a Micronaut 3.9.x application to Micronaut 4.0.1. The application contains a scheduled job using a fixed rate provided through configuration.

```java
@Scheduled(
      fixedRate = "${sales-system.offers.clean-up.fixed-rate}",
      condition = "#{!this.paused}")
@ReadOnly
public void deleteExpiredOfferLeads() { }
```

When running the tests or starting the application I get the following error message. For some reason, I currently do not yet understand, Micronaut does not resolve the expression `${sales-system.offers.clean-up.fixed-rate}`.

```
io.micronaut.scheduling.exceptions.SchedulerConfigurationException: Invalid @Scheduled definition for method: void deleteExpiredOfferLeads() - Reason: Invalid fixed rate definition: ${sales-system.offers.clean-up.fixed-rate}
	at io.micronaut.scheduling.processor.ScheduledMethodProcessor.lambda$process$3(ScheduledMethodProcessor.java:161)
	at java.base/java.util.Optional.orElseThrow(Optional.java:403)
	at io.micronaut.scheduling.processor.ScheduledMethodProcessor.process(ScheduledMethodProcessor.java:160)
	at io.micronaut.context.DefaultBeanContext.lambda$initializeContext$31(DefaultBeanContext.java:2032)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1423)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1311)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1841)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1806)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
```

I was wondering if someone knows how to tackle this one, since `ScheduledMethodProcessor` does not seem to have any expression evaluation features in `io.micronaut.scheduling.processor.ScheduledMethodProcessor#process`. Can anyone give some hints on this?

**Update 2023-07-27**

The issue seems to be related when using the `condition` attribute in the `@Scheduled` annotation. If I remove the `condition` parameter, everything works as with Micronaut 3.x

I have filed an [issue](https://github.com/micronaut-projects/micronaut-core/issues/9622) for this since I think this is bug.

**Comment by saw303** [comment]
**TL;DR** that's most likely a bug in Micronaut 4.0.0 and 4.0.1.

I'm closing this, since the rest of the discussion will be at https://github.com/micronaut-projects/micronaut-core/issues/9622
